File: codes/julia/003_conditionals.jl
#=
    Relational Operators / Comparison Operators:
    equal to (==)
    not equal to (!=)
    identical to (===) *also compares by location in memory.
    not identical to (!==) *also compares by location in memory.
    greater than (>)
    less than (<)
    greater than or equal to (>=)
    less than or equal to (<=)

    Logical Operators:
    Logical AND (&&)
    Logical OR (||)
    Logical NOT (!)
=#

println("\n# basic conditional control flow\n")

my_age = 10
println("my_age: ", my_age)
if (my_age > 24)
    println("you are old")
elseif (my_age > 17)
    println("you are young")
else
    println("you are under age")
end

println("\n# equal to (==)\n")

correct_answer = 100
my_answer = 100.0
println("my_answer: ", my_answer)
println("correct_answer: ", correct_answer)
println("(my_answer == correct_answer): ")
if (my_answer == correct_answer)
    println("Your answer is correct")
else
    println("Your answer is wrong")
end

println("\n# not equal to (!=)\n")

correct_answer = 100
my_answer = 25.0
println("my_answer: ", my_answer)
println("correct_answer: ", correct_answer)
println("(my_answer !== correct_answer): ")
if (my_answer !== correct_answer)
    println("Your answer is correct")
else
    println("Your answer is wrong")
end

println("\n# identical to (===)\n")

correct_answer = 100
my_answer = 100.0
println("my_answer: ", my_answer)
println("correct_answer: ", correct_answer)
println("(my_answer === correct_answer): ")
if (my_answer === correct_answer)
    println("Your answer is correct")
else
    println("Your answer is wrong")
end

println("\n# not identical to (!==)\n")

correct_answer = 100
my_answer = 25.0
println("my_answer: ", my_answer)
println("correct_answer: ", correct_answer)
println("(my_answer !== correct_answer): ")
if (my_answer !== correct_answer)
    println("Your answer is correct")
else
    println("Your answer is wrong")
end

println("\n# greater than (>)\n")

my_age = 70
println("my_age: ", my_age)
println("(my_age > 50): ")
if (my_age > 50) println("You are old") end

println("\n# less than (<)\n")

my_age = 16
println("my_age: ", my_age)
println("(my_age < 20): ")
if (my_age < 20) println("You are young") end

println("\n# greater than or equal to (>=)\n")

my_age = 73
println("my_age: ", my_age)
println("(my_age >= 65): ")
if (my_age >= 65) println("You are allowed to retire") end

println("\n# less than or equal to (<=)\n")

my_age = 14
println("my_age: ", my_age)
println("(my_age <= 16): ")
if (my_age <= 16) println("You are not allowed to drive") end

println("\n# Logical AND (&&)\n")

my_age = 17
println("my_age: ", my_age)
has_driving_license = false
println("has_driving_license: ", has_driving_license)
println("((my_age >= 17) && (has_driving_license === true)): ")
if ((my_age >= 17) && (has_driving_license === true)) println("You are allowed to drive") else println("You are not allowed to drive") end

my_age = 17
println("my_age: ", my_age)
has_driving_license = true
println("has_driving_license: ", has_driving_license)
println("((my_age >= 17) && (has_driving_license === true)): ")
if ((my_age >= 17) && (has_driving_license === true)) println("You are allowed to drive") else println("You are not allowed to drive") end

println("\n# Logical OR (||)\n")

my_age = 2
println("my_age: ", my_age)
println("((my_age <= 3) || (my_age >= 65)): ")
if ((my_age <= 3) || (my_age >= 65)) println("You should stay home") end

my_age = 89
println("my_age: ", my_age)
println("((my_age <= 3) || (my_age >= 65)): ")
if ((my_age <= 3) || (my_age >= 65)) println("You should stay home") end


println("\n# Logical NOT (!)\n")

can_drive = false
println("can_drive:", can_drive)
println("!canDrive:", !canDrive)

can_drive = true
println("can_drive:", can_drive)
println("!canDrive:", !canDrive)

File: codes/lua/003_conditionals.lua
--[[
    Relational Operators / Comparison Operators:
    equal to (==)
    not equal to (~=)
    greater than (>)
    less than (<)
    greater than or equal to (>=)
    less than or equal to (<=)

    Logical Operators:
    Logical AND (and)
    Logical OR (or)
    Logical NOT (not)
]]

function sprint(...)
    local parameters = {...}
    local result = ""
    for _, parameter in ipairs(parameters) do
        result = result .. tostring(parameter)
    end
    print(result)
end

print("\n-- basic conditional control flow\n")

my_age = 10
sprint("my_age: ", my_age)
if (my_age > 24) then
    print("you are old")
elseif (my_age > 17) then
    print("you are young")
else
    print("you are under age")
end

print("\n-- equal to (==)\n")

correct_answer = 100
my_answer = 100
sprint("my_answer: ", my_answer)
sprint("correct_answer: ", correct_answer)
print("(my_answer == correct_answer): ")
if (my_answer == correct_answer) then
    print("Your answer is correct")
else
    print("Your answer is wrong")
end

print("\n-- not equal to (~=)\n")

correct_answer = 100
my_answer = 25
sprint("my_answer: ", my_answer)
sprint("correct_answer: ", correct_answer)
print("(my_answer ~= correct_answer): ")
if (my_answer ~= correct_answer) then
    print("Your answer is correct")
else
    print("Your answer is wrong")
end

print("\n-- greater than (>)\n")

my_age = 70
sprint("my_age: ", my_age)
print("(my_age > 50): ")
if (my_age > 50) then print("You are old") end

print("\n-- less than (<)\n")

my_age = 16
sprint("my_age: ", my_age)
print("(my_age < 20): ")
if (my_age < 20) then print("You are young") end

print("\n-- greater than or equal to (>=)\n")

my_age = 73
sprint("my_age: ", my_age)
print("(my_age >= 65): ")
if (my_age >= 65) then print("You are allowed to retire") end

print("\n-- less than or equal to (<=)\n")

my_age = 14
sprint("my_age: ", my_age)
print("(my_age <= 16): ")
if (my_age <= 16) then print("You are not allowed to drive") end

print("\n-- Logical AND (and)\n")

my_age = 17
sprint("my_age: ", my_age)
has_driving_license = false
sprint("has_driving_license: ", has_driving_license)
print("((my_age >= 17) and (has_driving_license == true)): ")
if ((my_age >= 17) and (has_driving_license == true)) then print("You are allowed to drive") else print("You are not allowed to drive") end

my_age = 17
sprint("my_age: ", my_age)
has_driving_license = true
sprint("has_driving_license: ", has_driving_license)
print("((my_age >= 17) and (has_driving_license == true)): ")
if ((my_age >= 17) and (has_driving_license == true)) then print("You are allowed to drive") else print("You are not allowed to drive") end

print("\n-- Logical OR (or)\n")

my_age = 2
sprint("my_age: ", my_age)
print("((my_age <= 3) or (my_age >= 65)): ")
if ((my_age <= 3) or (my_age >= 65)) then print("You should stay home") end

my_age = 89
sprint("my_age: ", my_age)
print("((my_age <= 3) or (my_age >= 65)): ")
if ((my_age <= 3) or (my_age >= 65)) then print("You should stay home") end

print("\n-- Logical NOT (not)\n")

can_drive = false
sprint("can_drive: ", can_drive)
sprint("not can_drive: ", not can_drive)

can_drive = true
sprint("can_drive: ", can_drive)
sprint("not can_drive: ", not can_drive)
